src/components/ImageUploader.js

```diff
@@ -0,0 +1,257 @@
+const React = require('react');
+const Radium = require('radium');
+const numeral = require('numeral');
+
+const StyleConstants = require('../constants/Style');
+
+const Icon = require('./Icon');
+const Button = require('./Button');
+
+const ImageUploader = React.createClass({
+  propTypes: {
+    allowedFileTypes: React.PropTypes.array,
+    color: React.PropTypes.string,
+    fileBeingDragged: React.PropTypes.bool,
+    maxFileSize: React.PropTypes.number,
+    onFileError: React.PropTypes.func,
+    onFileRemove: React.PropTypes.func,
+    onFileSuccess: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      allowedFileTypes: ['png', 'jpg', 'gif'],
+      color: StyleConstants.Colors.PRIMARY,
+      fileBeingDragged: false,
+      maxFileSize: 150, //in kB
+      onFileError: () => {},
+      onFileRemove: () => {}
+    };
+  },
+
+  getInitialState () {
+    return {
+      fileContent: null,
+      invalidMessage: null,
+      uploadedFile: null
+    };
+  },
+
+  _onDropzoneClick () {
+    this.refs.hiddenInput.click();
```

Comment:
Per React docs, using a ref string is "mostly legacy at this point" (https://facebook.github.io/react/docs/more-about-refs.html#the-ref-string-attribute). Could we adjust this to use the ref callback approach?